python/test/unit/fem/test_petsc_nonlinear_assembler.py

```diff
@@ -492,30 +344,18 @@ def monolithic_solve():
                 dirichletbc(u1_bc, bdofsW1_V1, W.sub(1)),
             ]
 
-            Jmat = create_matrix(J)
-            Fvec = create_vector(F)
-
-            snes = PETSc.SNES().create(MPI.COMM_WORLD)
-            snes.setTolerances(rtol=1.0e-15, max_it=10)
-
-            problem = NonlinearPDE_SNESProblem(F, J, U, bcs)
-            snes.setFunction(problem.F_mono, Fvec)
-            snes.setJacobian(problem.J_mono, J=Jmat, P=None)
-
+            snes_options = {"snes_rtol": 1.0e-15, "snes_max_it": 10}
             U.sub(0).interpolate(initial_guess_u)
             U.sub(1).interpolate(initial_guess_p)
 
-            x = create_vector(F)
-            x.array[:] = U.x.petsc_vec.array_r
-
-            snes.solve(None, x)
-            assert snes.getKSP().getConvergedReason() > 0
-            assert snes.getConvergedReason() > 0
+            solver = dolfinx.nls.petsc.SNESSolver(
+                F, U, J=J, bcs=bcs, assembly_type=dolfinx.fem.AssemblyType.default
+            )
+            solver.set_options(snes_options)
+            x, converged_reason, _ = solver.solve()
+            assert converged_reason > 0
+            solver.replace_solution(U, x)
```

Comment:
We should discuss where this should go, inside or outside solve, now that there is a lower level path as well.